ostd/src/mm/page/allocator.rs

```diff
@@ -13,9 +13,38 @@ use log::info;
 use spin::Once;
 
 use super::{cont_pages::ContPages, meta::PageMeta, Page};
-use crate::{boot::memory_region::MemoryRegionType, mm::PAGE_SIZE, sync::SpinLock};
+use crate::{
+    boot::memory_region::MemoryRegionType,
+    mm::{stat::allocated, PAGE_SIZE},
+    sync::SpinLock,
+};
 
-pub(in crate::mm) static PAGE_ALLOCATOR: Once<SpinLock<FrameAllocator>> = Once::new();
+pub(in crate::mm) struct CountingFrameAllocator {
```

Comment:
A little bit hilarious since FrameAllocator have the total and allocated fields [source](https://docs.rs/buddy_system_allocator/latest/src/buddy_system_allocator/frame.rs.html#33-40), but no public APIs to get it. It is OK for you to temporarily do so.